You're a data analyst for an e-commerce company, and you have a dataset of customer orders. The dataset contains the following variables:
`Order ID`: a unique identifier for each order
`Customer ID`: a unique identifier for each customer
`Order Date`: the date the order was placed
`Product`: the product ordered
`Quantity`: the quantity of the product ordered
`Total Cost`: the total cost of the order
*Task:*

1. Create a Pandas DataFrame from the following data:
data = {'Order ID': [1, 2, 3, 4, 5],
'Customer ID': [101, 102, 101, 103, 102],
'Order Date': ['2022-01-01', '2022-01-05', '2022-01-10', '2022-01-15', '2022-01-20'],
'Product': ['Product A', 'Product B', 'Product A', 'Product C', 'Product B'],
'Quantity': [2, 3, 1, 4, 2],
'Total Cost': [100, 150, 50, 200, 100]}
2. Calculate the total revenue for each product.
3. Find the customer who placed the most orders.
4. Calculate the average order value for each customer.
5. Find the product with the highest total quantity sold.

*Deliverables:*
A Pandas DataFrame showing the total revenue for each product.
The customer ID of the customer who placed the most orders.
A Pandas DataFrame showing the average order value for each customer.
The product with the highest total quantity sold.
*Tips:*
Use the `groupby` function to group the data by product and calculate the total revenue.
Use the `value_counts` function to find the customer who placed the most orders.
Use the `groupby` function to group the data by customer and calculate the average order value.
Use the `sum` function to calculate the total quantity sold for each product.

 Subtask 1: Create the DataFrame

In [ ]:
import pandas as pd
data = {
    'Order ID': [1, 2, 3, 4, 5],
    'Customer ID': [101, 102, 101, 103, 102],
    'Order Date': ['2022-01-01', '2022-01-05', '2022-01-10', '2022-01-15', '2022-01-20'],
    'Product': ['Product A', 'Product B', 'Product A', 'Product C', 'Product B'],
    'Quantity': [2, 3, 1, 4, 2],
    'Total Cost': [100, 150, 50, 200, 100]
}
df = pd.DataFrame(data)
df


Order ID  Customer ID  Order Date    Product  Quantity  Total Cost
0         1          101  2022-01-01  Product A         2         100
1         2          102  2022-01-05  Product B         3         150
2         3          101  2022-01-10  Product A         1          50
3         4          103  2022-01-15  Product C         4         200
4         5          102  2022-01-20  Product B         2         100

subtask 2 : calculate total revenue for each product
steps :


1.   first group and sum
2.   then convert to dataframe
3.   reset_index for adding index







In [ ]:
revenue_per_product = df.groupby('Product')['Total Cost'].sum().reset_index().sort_values(by='Total Cost', ascending=False)
revenue_per_product


Product  Total Cost
1  Product B         250
2  Product C         200
0  Product A         150

subtask 3: find customer who placed the most orders

steps :
1] value_counts() -> calculate how many times each customer placed an order
2] idxmax()->find customer id with highest no of order






In [ ]:
most_orders = df['Customer ID'].value_counts().idxmax()
print("Customer with most orders:", most_orders)

Customer with most orders: 101


subtask 4 : calculate average order value for each customer

In [ ]:
avg_order_value = df.groupby('Customer ID')['Total Cost'].mean().reset_index()
avg_order_value.columns = ['Customer ID', 'Average Order Value']
avg_order_value


Customer ID  Average Order Value
0          101                 75.0
1          102                125.0
2          103                200.0

subtask 5 : find product with highest total quatity sold

In [ ]:
quantity_per_product = df.groupby('Product')['Quantity'].sum()
top_product = quantity_per_product.idxmax()
print("Product with highest total quantity sold:", top_product)


Product with highest total quantity sold: Product B


key finding=>


*   The dataset contains 5 orders from 3 unique customers across 3 products and 20 days.
*   Product B generated the highest total revenue (₹250), followed by Product C and A.
*   Customer ID 101 placed the most orders.
*   Customer 103 had the highest average order value (₹200).
*   Product C was sold the most by quantity (4 units).








insight=>


*   The structured data includes essential fields like Customer ID, Order Date, Product, Quantity, and Total Cost, which enables product-wise, customer-wise, and time-wise analysis.
*   Product B is a strong revenue contributor and should be prioritized in future promotions.


*   Repeat customers like 101 indicate strong engagement and are ideal candidates for loyalty rewards.
*   This customer likely buys high-ticket items — they could be targeted with premium product offers.


*   Product C has strong demand — consider it for restocking and bundle offers.








